fapolicy_analyzer/ui/profiler_page.py

```diff
@@ -0,0 +1,109 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import glob
+import logging
+import gi
+import os
+
+from time import sleep
+from fapolicy_analyzer.ui.store import get_system_feature
+from fapolicy_analyzer.ui.ui_page import UIAction, UIPage
+from fapolicy_analyzer.ui.ui_widget import UIConnectedWidget
+
+gi.require_version("Gtk", "3.0")
+from gi.repository import Gtk
+
+
+class ProfilerPage(UIConnectedWidget, UIPage):
+    def __init__(self):
+        UIConnectedWidget.__init__(self, get_system_feature())
+
+        actions = {
+            "start": [
+                UIAction(
+                    "Start",
+                    "Start Test",
+                    "media-playback-start",
+                    {"clicked": self.on_test_activate},
+                    sensitivity_func=self.start_button_sensitivity
+                )
+            ],
+            "stop": [
+                UIAction(
+                    "Stop",
+                    "Stop Test",
+                    "media-playback-stop",
+                    {},
+                    sensitivity_func=self.stop_button_sensitivity
+                )
+            ]
+        }
+
+        UIPage.__init__(self, actions)
+
+        self._fapd_mgr = None
+        self._fapd_profiler = None
+        self.running = False
+
+    def stop_button_sensitivity(self):
+        return self.running
+
+    def start_button_sensitivity(self):
+        return not self.running
+
+    def get_text(self):
+        entryDict = {
+            "executeText": self.get_object("executeEntry").get_text(),
+            "argText": self.get_object("argEntry").get_text(),
+            "userText": self.get_object("userEntry").get_text(),
+            "dirText": self.get_object("dirEntry").get_text(),
+            "envText": self.get_object("envEntry").get_text(),
+        }
+
+        return entryDict
+
+    def display_log_output(self):
+        text_display = self.get_object("profilerOutput")
+        buff = Gtk.TextBuffer()
+        work_dir = self.get_text()["dirText"]
+        if work_dir is not None:
+            files = glob.glob(work_dir + "/*")
+            latest_file = max(files, key=os.path.getctime)
+            time = "_".join(latest_file.split("_")[2:5])
+            run_files = [f for f in files if time in f]
+
+            for run_file in run_files:
+                buff.insert_markup(buff.get_end_iter(), f"<b>{run_file}</b>", -1)
+                with open(run_file, "r") as f:
+                    lines = f.readlines()
```

Comment:
You need to make sure you handle any exceptions during the read here.  Think of @jw3 testing with `deny perm=any all : all`.